### DISCUSSION TITLE
Is it possible to pass input through an egui window?

### DISCUSSION THREAD

**Discussion by popcar2** [emilk/egui#1656] [Q&A] [answered]
Is it possible to pass input (mouse/keyboard) through an egui window such that it can reach applications. I'm making an overlay for a game and while I managed to make it transparent and always on top, I can't click anything under it. 

There was another thread talking about this https://github.com/emilk/egui/discussions/354 but it never seemed to have gotten resolved. Has anyone managed to pass input through an egui window? If not, does anyone know another good library I can do this with? Thanks in advance.

It seems [wants_pointer_input()](https://docs.rs/egui/latest/egui/struct.Context.html#method.wants_pointer_input) and [wants_keyboard_input()](https://docs.rs/egui/latest/egui/struct.Context.html#method.wants_keyboard_input) can tell when the mouse/keyboard are on top of a widget or not and returns false when hovering over the transparent window background. Maybe something is doable with this?

![LBVAoljZWd](https://user-images.githubusercontent.com/16920817/169650767-37dc657d-ba24-4d02-b274-ea9bbfe44ab5.png)

**Comment by coderedart** [comment, marked as answer]
Aye. Window passthrough has not yet landed in glfw stable yet. So, I have a custom fork which you can use https://github.com/coderedart/egui_overlay
